experimental/list_formatter/src/list_formatter.rs

```diff
@@ -139,62 +153,54 @@ mod tests {
     }
 
     #[test]
-    fn test_format() {
+    fn test_writeable() {
         let formatter = formatter(Width::Wide);
-        assert_eq!(formatter.format(&VALUES[0..0]), "");
-        assert_eq!(formatter.format(&VALUES[0..1]), "one");
-        assert_eq!(formatter.format(&VALUES[0..2]), "$one;two+");
-        assert_eq!(formatter.format(&VALUES[0..3]), "@one:two.three!");
-        assert_eq!(formatter.format(&VALUES[0..4]), "@one:two,three.four!");
-        assert_eq!(formatter.format(VALUES), "@one:two,three,four.five!");
+        assert_writeable_eq!(formatter.format(&VALUES[0..0]).to_writeable(), "");
+        assert_writeable_eq!(formatter.format(&VALUES[0..1]).to_writeable(), "one");
+        assert_writeable_eq!(formatter.format(&VALUES[0..2]).to_writeable(), "$one;two+");
+        assert_writeable_eq!(
+            formatter.format(&VALUES[0..3]).to_writeable(),
+            "@one:two.three!"
+        );
+        assert_writeable_eq!(
+            formatter.format(&VALUES[0..4]).to_writeable(),
+            "@one:two,three.four!"
+        );
+        assert_writeable_eq!(
+            formatter.format(VALUES).to_writeable(),
+            "@one:two,three,four.five!"
+        );
     }
 
     #[test]
-    fn test_format_to_parts() {
+    fn test_fmt_writeable() {
         let formatter = formatter(Width::Wide);
 
-        assert_eq!(formatter.format_to_parts(&VALUES[0..0]).as_ref(), "");
-        assert_eq!(formatter.format_to_parts(&VALUES[0..1]).as_ref(), "one");
         assert_eq!(
-            formatter.format_to_parts(&VALUES[0..2]).as_ref(),
-            "$one;two+"
+            format!("{:?}", formatter.format(VALUES).writeable_to_fmt_string()),
+            "@one:two,three,four.five!\n\
+             ┏┏━━┏┏━━┏┏━━━━┏┏━━━┏┏━━━┏\n\
```

Comment:
Nit (optional): Although I like the clever use of the Unicode Box Drawing code points, they do not line up for me on GitHub.  I'd probably err on the side of using ASCII art since the primary use case for this is to be read in a monospace layout and frequently on GitHub.

Reply:
What do you mean they don't line up? I get spaces between lines, but horizontally everything is correct.

Reply:
This is what it looks like to me on Chrome OS:

![image](https://user-images.githubusercontent.com/1145762/148634043-6deba6b3-3b1a-47bc-9a72-a3a044a76791.png)

Reply:
I think we should file a bug with Chrome OS. It works on Chrome on macOS, so it's not something GitHub can fix, and it's not the first time ChromeOS's rendering is broken.